src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -54,16 +56,16 @@ internal partial class ServiceConnection : ServiceConnectionBase
                                  HubServiceEndpoint endpoint,
                                  IServiceMessageHandler serviceMessageHandler,
                                  ServiceConnectionType connectionType = ServiceConnectionType.Default,
-                                 int closeTimeOutMilliseconds = DefaultCloseTimeoutMilliseconds,
-                                 bool enableConnectionMigration = false) :
-            base(serviceProtocol, serverId, connectionId, endpoint, serviceMessageHandler, connectionType, loggerFactory?.CreateLogger<ServiceConnection>())
+                                 ServerConnectionMigrationLevel migrationLevel = ServerConnectionMigrationLevel.Off,

```

Comment:
> = ServerConnectionMigrationLevel.Off [](start = 78, length = 37)

probably make default `Off` assigned in `ServiceConnectionFactory` instead of here?

Reply:
I can do both. And it's necessary here since the previous param is already optional.

Reply:
Move it before optional parameter? I notice that if make the parameter optional in ServiceConnectionBase/ServiceConnectionFactory could reduce the assign value in AspNet/Management side which seems not supported the feature yet. So just wonderding if we should do like that. 

---
In reply to: [396884644](https://github.com/Azure/azure-signalr/pull/857#discussion_r396884644) [](ancestors = 396884644)